modules/data_release/ajax/AddPermission.php

```diff
@@ -12,25 +12,92 @@
  *  @link     https://github.com/aces/Loris
  */
 
-$DB =& Database::singleton();
+$user     = \User::singleton();
+$factory  = \NDB_Factory::singleton();
+$settings = $factory->settings();
+$baseURL  = $settings->getBaseURL();
+$DB       = $factory->database();
 
-if ($_POST['action'] == 'addpermission') {
-    $userid          = $_POST['userid'];
-    $data_release_id = $_POST['data_release_id'];
-    $success         = $DB->insert(
-        'data_release_permissions',
-        array(
-         'userid'          => $userid,
-         'data_release_id' => $data_release_id,
-        )
-    );
+if ($_POST['action'] == 'addpermission' && $user->hasPermission('superuser')) {
+    if (!empty($_POST['data_release_id']) && empty($_POST['data_release_version'])) {
+        $userid          = $_POST['userid'];
+        $data_release_id = $_POST['data_release_id'];
+        $success         = $DB->insert(
+            'data_release_permissions',
+            array(
+             'userid'          => $userid,
+             'data_release_id' => $data_release_id,
+            )
+        );
+    } elseif (empty($_POST['data_release_id'])
+        && !empty($_POST['data_release_version'])
+    ) {
+        $userid = $_POST['userid'];
+        $data_release_version = $_POST['data_release_version'];
 
-    $factory  = NDB_Factory::singleton();
-    $settings = $factory->settings();
+        $IDs = $DB->pselectCol(
+            "SELECT id FROM data_release WHERE "
+            . "version=:data_release_version",
+            array('data_release_version' => $data_release_version)
+        );
 
-    $baseURL = $settings->getBaseURL();
+        foreach ($IDs as $ID) {
+            $success = $DB->insert(
+                'data_release_permissions',
+                array(
+                 'userid'          => $userid,
+                 'data_release_id' => $ID['id'],
+                )
+            );
+        }
+    }
```

Comment:
If I only enter a username, I receive a 302 and get redirected to `/data_release/?addpermissionSuccess=true`
Are version and id required and mutually exclusive?

Reply:
Looking at my code, yeah it seems to be...and good point there should be more error handling/feedback!